app/javascript/profileDropdown/flagButton.js

```diff
@@ -58,10 +34,49 @@ export function initFlag() {
             flagButton.innerHTML = `Flag ${profileUserName}`;
           }
         })
-        .catch((e) => window.alert(`Something went wrong: ${e}`));
+        .catch((e) => {
+          Honeybadger.notify(
+            isUserFlagged ? 'Unable to unflag user' : 'Unable to flag user',
+            profileUserId,
+          );
+          window.alert(`Something went wrong: ${e}`);
+        });
     }
   }
 
   flagButton.addEventListener('click', flag);
 }
+
+/**
+ * Adds a flag button visible only to trusted users on profile pages.
+ * @function initFlag
+ * @returns {(void|undefined)} This function has no useable return value.
+ */
+
+export function initFlag() {
+  const flagButton = document.getElementById(
+    'user-profile-dropdownmenu-flag-button',
+  );
+
+  if (!flagButton) {
+    // button not always present when this is called
+    return;
+  }
+
+  getUserDataAndCsrfToken().then(() => {
+    const user = userData();
+    if (!user) {
+      flagButton.remove;
```

Comment:
Refactor (blocking): The function never gets called. It needs to be invoked.

```suggestion
      flagButton.remove();
```

Reply:
Thanks for catching this - I think it wasn't caught in testing since `!user` was no longer true when the flag button was found, (there's a backend guard in place to only include the actions to logged in users) - so that's reassuring.

The function call is possibly unneeded now (but fixed).